community/kernel/src/main/java/org/neo4j/kernel/impl/locking/community/CommunityLockClient.java

```diff
@@ -44,129 +62,243 @@ public CommunityLockClient( LockManagerImpl manager )
     @Override
     public void acquireShared( Locks.ResourceType resourceType, long resourceId )
     {
-        PrimitiveLongObjectMap<LockResource> localLocks = localShared( resourceType );
-        LockResource resource = localLocks.get( resourceId );
-        if( resource != null )
+        if ( !stateHolder.incrementActiveClients() )
         {
-            resource.acquireReference();
+            throw new LockClientAlreadyClosed( String.format( "%s is already closed", this ) );
+        }
+        try
+        {
+            PrimitiveLongObjectMap<LockResource> localLocks = localShared( resourceType );
+            LockResource resource = localLocks.get( resourceId );
+            if ( resource != null )
+            {
+                resource.acquireReference();
+            }
+            else
+            {
+                resource = new LockResource( resourceType, resourceId );
+                if ( manager.getReadLock( resource, lockTransaction ) )
+                {
+                    localLocks.put( resourceId, resource );
+                }
```

Comment:
Currently lock clients closed from either `IdleChannelReaper` or `TimedRepository` in case of inactivities. (`KernelTransactionImplementation` closing locks in the same thread where it was used and only when its disposed so its not our case ) That kind of situation can occur when we losing some node from cluster. And since it's already non active and connection is closed there will be nobody on another side.